<launch>
    <node name="nv_producer" pkg="nvidia_to_ros" type="nv_producer" output="screen" required="true" />
    <node name="nv_consumer" pkg="nvidia_to_ros" type="nv_consumer" output="screen" required="true">
      <param name="visualize" value="true" />
      <rosparam param="input_topics">[/nv_image1, /nv_image2]</rosparam>
      <rosparam param="output_topics">[/camera_image1, /camera_image2]</rosparam> 
    </node>

    
  <node name="nv_transform" pkg="nvidia_to_ros" type="nv_transform" output="screen" >
    <rosparam param="input_topics">[/nv_image1]</rosparam>
    <rosparam param="output_topics">[/transformed_image1]</rosparam>


    <param name="convert_to_gray" value="false" />
    <param name="new_width" value="-1" />
    <param name="new_height" value="-1" />
    <param name="crop_src_rect_top" value="-1" />
    <param name="crop_src_rect_left" value="-1" />
    <param name="crop_src_rect_width" value="-1" />
    <param name="crop_src_rect_height" value="-1" />
    <param name="crop_dst_rect_top" value="-1" />
    <param name="crop_dst_rect_left" value="-1" />
    <param name="crop_dst_rect_width" value="-1" />
    <param name="crop_dst_rect_height" value="-1" />
    
    <!--
	0 NvBufferTransform_None
	1 NvBufferTransform_Rotate90  (ccw)
	2 NvBufferTransform_Rotate180 (ccw)
	3 NvBufferTransform_Rotate270 (ccw)
	4 NvBufferTransform_FlipX
	5 NvBufferTransform_FlipY
	6 NvBufferTransform_Transpose    (flip x, then rotate 90 ccw)
	7 NvBufferTransform_InvTranspose (flip x, then rotate 90 cw)
    -->
    <param name="flip" value="-1" />
    
    <!--
	0 NvBufferTransform_Filter_Nearest
	1 NvBufferTransform_Filter_Bilinear
	2 NvBufferTransform_Filter_5_Tap
	3 NvBufferTransform_Filter_10_Tap
	4 NvBufferTransform_Filter_Smart
	5 NvBufferTransform_Filter_Nicest
    -->
    <param name="filter" value="-1" />
  </node>

  <node name="rqt_camera1" pkg="rqt_image_view" type="rqt_image_view" args="/camera_image1" />
  <node name="rqt_camera2" pkg="rqt_image_view" type="rqt_image_view" args="/camera_image2" />
  <node name="rqt_transformed" pkg="rqt_image_view" type="rqt_image_view" args="/transformed_image1" />
</launch>
